src/test/utils.ts

```diff
@@ -0,0 +1,177 @@
+ *   profileScope/profileName.supr.ast.json  - compiled profile source
+ *   profileScope/profileName.json           - profile info
+ */
+export async function mockResponsesForProfile(
+  server: Mockttp,
+  profile: string,
+  path = joinPath('fixtures', 'profiles')
+): Promise<void> {
+  const basePath = joinPath(path, profile);
+  const profileInfo = JSON.parse(
+    (await readFile(basePath + '.json')).toString()
+  );
+  const profileSource = (
+    await readFile(basePath + EXTENSIONS.profile.source)
+  ).toString();
+  const profileAST = JSON.parse(
+    (await readFile(basePath + EXTENSIONS.profile.build)).toString()
+  );
+  await server
+    .get('/' + profile)
+    .withHeaders({ Accept: ContentType.JSON })
+    .thenJson(200, profileInfo);
+  await server
+    .get('/' + profile)
+    .withHeaders({ Accept: ContentType.PROFILE })
+    .thenReply(200, profileSource, { ContentType: ContentType.PROFILE });
+  await server
+    .get('/' + profile)
+    .withHeaders({ Accept: ContentType.AST })
+    .thenJson(200, profileAST, { 'Content-Type': ContentType.AST });
+}
+
+/**
+ * Mocks HTTP responses for a provider
+ *
+ * expects following files in specified path (default fixtures/providers)
+ *   provider.json           - provider info
+ */
+export async function mockResponsesForProvider(
+  server: Mockttp,
+  provider: string,
+  path = joinPath('fixtures', 'providers')
+): Promise<void> {
+  const basePath = joinPath(path, provider);
+  const providerInfo = JSON.parse(
+    (await readFile(basePath + '.json')).toString()
+  );
+
+  await server
+    .get('/providers/' + provider)
+    .withHeaders({ Accept: ContentType.JSON })
+    .thenJson(200, providerInfo);
+}
+
+/**
+ * Executes the Superface CLI binary
+ *
+ * @export
+ * @param {string} directory - the directory in which the process runs
+ * @param {string[]} args - arguments of the process
+ * @param {string} apiUrl - the API URL (to be overriden with mock)
+ * @param {string [] | undefined} inputs - the inquier prompt inputs
+ * @param {boolean} debug - pass child process stdout to console
+ * @param {NodeJS.ProcessEnv} [env] - any additional environment variables
+ * @returns  {Promise<string>} - result is concatenated stdout
+ */
+export async function execCLI(
+  directory: string,
+  args: string[],
+  apiUrl: string,
+  inputs?: string[],
+  env?: NodeJS.ProcessEnv,
+  debug?: boolean
+): Promise<{ stdout: string }> {
+  const timeout = 100,
+    maxTimeout = 20000;
+  const CLI = joinPath('.', 'bin', 'superface');
+  const bin = relative(directory, CLI);
+
+  const childProcess = execFile(bin, args, {
+    cwd: directory,
+    env: { ...process.env, ...env, SUPERFACE_API_URL: apiUrl },
+  });
+
+  childProcess.stdin?.setDefaultEncoding('utf-8');
+
+  let currentInputTimeout: NodeJS.Timeout, killIOTimeout: NodeJS.Timeout;
+
+  // Creates a loop to feed user inputs to the child process in order to get results from the tool
+  const loop = (inputs: string[]) => {
```

Comment:
don't shadow names pls. add rule to eslint pls, it's not on for some reason: https://eslint.org/docs/rules/no-shadow